File: CS/04_Graphs/DrawFilledRectangles.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;

namespace DrawFilledRectangles
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\DrawingTemplate.pdf");
            //Create one page
            PdfPageBase page = pdf.Pages[0];

            //save graphics state
            PdfGraphicsState state = page.Canvas.Save();

            //Set rectangle display location and size
            int x = 200;
            int y = 300;
            int width = 200;
            int height = 120;
            //Create one page and brush
            PdfPen pen = new PdfPen(Color.Black, 1f);
            PdfBrush brush = new PdfSolidBrush(Color.OrangeRed);
            //Draw a filled rectangle
            page.Canvas.DrawRectangle(pen, brush, new Rectangle(new Point(x, y), new Size(width, height)));

            //restor graphics
            page.Canvas.Restore(state);

            String result = "DrawFilledRectangles_out.pdf";

            //Save the document
            pdf.SaveToFile(result);
            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/04_Graphs/DrawFilledRectangles.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace DrawFilledRectangles
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PDF document
			Dim pdf As New PdfDocument()

			' Load a PDF file from the specified path
			pdf.LoadFromFile("..\..\..\..\..\..\Data\DrawingTemplate.pdf")

			' Get the first page of the PDF document
			Dim page As PdfPageBase = pdf.Pages(0)

			' Save the current graphics state of the page
			Dim state As PdfGraphicsState = page.Canvas.Save()

			' Define the coordinates and dimensions for the rectangle
			Dim x As Integer = 200
			Dim y As Integer = 300
			Dim width As Integer = 200
			Dim height As Integer = 120

			' Create a PDF pen with black color and thickness of 1
			Dim pen As New PdfPen(Color.Black, 1.0F)

			' Create a PDF brush with orange-red color
			Dim brush As PdfBrush = New PdfSolidBrush(Color.OrangeRed)

			' Draw a rectangle on the page using the specified pen and brush, with the defined dimensions
			page.Canvas.DrawRectangle(pen, brush, New Rectangle(New Point(x, y), New Size(width, height)))

			' Restore the previous graphics state of the page
			page.Canvas.Restore(state)

			' Specify the file name for the resulting PDF
			Dim result As String = "DrawFilledRectangles_out.pdf"

			' Save the modified PDF document to the specified file
			pdf.SaveToFile(result)

			' Close the document
			pdf.Close()

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub
		Private Sub PDFDocumentViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace
